### ISSUE TITLE
Feature Request: displaying as bit pattern

### ISSUE THREAD

**Issue by tnugent97** [rwpenney/spfpm#3]
Hi there,

I think it would be a good idea to have a function that returns a string of the bit representation of the fixed point number. 
That was the library can be used to generate fixed point bit patterns for hardware designs.

**Comment by rwpenney** [comment]
Sounds like a good idea, thanks.
I'll start experimenting soon.

**Comment by rwpenney** [comment]
There's now a (prototype) method 'toBinaryString()' in FXnum, which may do what you need. This also allows printing as hexadecimal and octal fixed-point representations.

There are some subtleties in representing negative numbers, which requires this routine to guess how many bits to use to hold the integer part of the fixed-point number unless this is given by the 'integer_bits' property of the FXfamily.

I'll need to do some more tidying and testing before doing a formal release, but any initial feedback would be welcome.

**Comment by rwpenney** [comment]
I believe this issue is resolved with the release of spfpm-1.4, which includes the `FXnum.toBinaryString()` method. As an illustration, you can now do things like:
```python
fam = FixedPoint.FXfamily(512)
for b in range(1,5):
    print(fam.log2.toBinaryString(b)[:60])
```
to get log(2) printed to 60-digit resolution in binary, base-4, octal and hexadecimal, as follows:
```
0.1011000101110010000101111111011111010001110011110111100110
0.2301130201133313310130331321222330213203230321200003330233
0.5427102775750717363257117073163000771366536403632062316246
0.b17217f7d1cf79abc9e3b39803f2f6af40f343267298b62d8a0d175b8b
```
